source/fundamentals/crud/query-document.txt

```diff
@@ -0,0 +1,192 @@
+    { "_id": 2, "color": "purple", "qty": 8, "vendor": ["B", "D", "F"], "rating": 5 }
+    { "_id": 7, "color": "green", "qty": 8, "vendor": ["C", "E"], "rating": 7 }
+
+.. _query-logical:
+
+Logical Operators
+-----------------
+
+Logical operators query data using logic applied to the results of
+field-level operators. Common logical operators include ``and()`` where
+all operators must be true, and ``or()`` where at least one of the
+operators must be true.
+
+The following example uses the and-operator to match documents where
+the value of the ``qty`` is less than or equal to ``5`` and the
+value of the ``color`` is not ``"pink"`` in the ``paint_purchases``
+collection: 
+
+.. literalinclude:: /includes/fundamentals/code-snippets/Query.java
+   :language: java
+   :dedent:
+   :start-after: begin logicalFilter
+   :end-before: end logicalFilter
+
+The following shows the output of the query using the operator specified
+above:
+
+.. code-block:: json
+   :copyable: false
+
+    { "_id": 3, "color": "blue", "qty": 5, "vendor": ["A", "E"] }
+    { "_id": 5, "color": "yellow", "qty": 4, "vendor": ["A", "B"] }
+
+.. _query-arrays:
+
+Array Operators
+---------------
+
+Array operators query data based on the value or quantity of elements in
+an array field. 
+
+The following example uses the size operator to match documents where
+the size of the ``vendor`` list is ``3`` in the ``paint_purchases``
+collection: 
+
+.. literalinclude:: /includes/fundamentals/code-snippets/Query.java
+   :language: java
+   :dedent:
+   :start-after: begin arrayFilter
+   :end-before: end arrayFilter
+
+The following shows the output of the query using the operator specified
+above:
+
+.. code-block:: json
+   :copyable: false
+
+    { "_id": 2, "color": "purple", "qty": 8, "vendor": ["B", "D", "F"], "rating": 5 }
+    { "_id": 8, "color": "black", "qty": 7, "vendor": ["A", "C", "D"] }
+
+.. _query-elements:
+
+Element Operators
+-----------------
+
+Element operators query data based on the presence or type of a field.  
+
+The following example uses the exists operator to match documents that
+have a ``rating`` in the ``paint_purchases`` collection: 
+
+.. literalinclude:: /includes/fundamentals/code-snippets/Query.java
+   :language: java
+   :dedent:
+   :start-after: begin elementFilter
+   :end-before: end elementFilter
+
+The following shows the output of the query using the operator specified
+above:
+
+.. code-block:: json
+   :copyable: false
+
+    { "_id": 2, "color": "purple", "qty": 8, "vendor": ["B", "D", "F"], "rating": 5 }
+    { "_id": 4, "color": "white", "qty": 6, "vendor": ["D"], "rating": 9 }
+    { "_id": 7, "color": "green", "qty": 8, "vendor": ["C", "E"], "rating": 7 }
+
+.. _query-evaluation:
+
+Evaluation Operators
+--------------------
+
+Evaluation operators query data on higher level logic, like regex
+and text searches. 
+
+The following example uses the regex operator to match documents that
```

Comment:
suggestion: I think it'd be more clear to say "The following example uses the ``Filters.regex`` method..." since they are using methods in the driver rather than specifying operators to the server. This applies throughout.